Source/Core/VideoCommon/HiresTextures.h

```diff
@@ -29,31 +30,20 @@ class HiresTexture
 
   static std::string GenBaseName(const TextureInfo& texture_info, bool dump = false);
 
-  static u32 CalculateMipCount(u32 width, u32 height);
-
   ~HiresTexture();
 
   AbstractTextureFormat GetFormat() const;
   bool HasArbitraryMipmaps() const;
 
-  struct Level
-  {
-    std::vector<u8> data;
-    AbstractTextureFormat format = AbstractTextureFormat::RGBA8;
-    u32 width = 0;
-    u32 height = 0;
-    u32 row_length = 0;
-  };
-  std::vector<Level> m_levels;
+  VideoCommon::CustomTextureData& GetData() { return m_data; }
+  const VideoCommon::CustomTextureData& GetData() const { return m_data; }
 
 private:
   static std::unique_ptr<HiresTexture> Load(const std::string& base_filename, u32 width,
                                             u32 height);
-  static bool LoadDDSTexture(HiresTexture* tex, const std::string& filename);
-  static bool LoadDDSTexture(Level& level, const std::string& filename, u32 mip_level);
-  static bool LoadTexture(Level& level, const std::vector<u8>& buffer);
   static void Prefetch();
 
   HiresTexture() = default;
   bool m_has_arbitrary_mipmaps = false;
+  VideoCommon::CustomTextureData m_data;
```

Comment:
While you're here... can you swap these two members (so the memory layout is the same as before, probably doesn't matter but y'know) and put an empty line between the constructor and data?

Reply:
Whoops, apologies.  Fixing...